### PR TITLE
improvement(views): Highlights View widget

### PR DESCRIPTION
**Pull request by soyacz**
Introducing Highlights View Widget that should aid Views with important messages, action items and allow commenting upon them. Originally, intended for performance view dashboard, where important messages and action items are discussed over weekly performance regression tests.

With this widget, users may:
* create messages (highlights) for given view
* create action items and assign Assignee, mark item as done upon completion
* comment each highlight/action item
* archive hightlights/action items to hide it from the view
* unarchive it to bring it back

More info in ref issue.

refs: https://github.com/scylladb/argus/issues/485

### CODE REVIEW

**Comment by soyacz** [comment]
Demo:
https://github.com/user-attachments/assets/88b16a20-8d8f-4dc0-9001-43053fe0d57a

**Comment by soyacz** [comment]
@juliayakovlev can you take a look at the demo and let me know WDYT?

**Comment by juliayakovlev** [comment]
> @juliayakovlev can you take a look at the demo and let me know WDYT?

Lukasz, Where can I see it? I mean not demo, in the Argus. Is it possible?

**Comment by soyacz** [comment]
> > @juliayakovlev can you take a look at the demo and let me know WDYT?
> 
> Lukasz, Where can I see it? I mean not demo, in the Argus. Is it possible?

Not yet, I didn't deploy it and can't do it on staging as it will modify View on prod. Demo is in linked video.

**Comment by juliayakovlev** [comment]
Commonly it looks Ok. I just do not understand where it will be located.

**Comment by soyacz** [comment]
> Commonly it looks Ok. I just do not understand where it will be located.

Somewhere around the top of https://argus.scylladb.com/view/performance-regression-tests

**Comment by soyacz** [comment]
I adjusted the code to support multiple highlight widgets (and it even works).

**Comment by fruch** [comment]
@soyacz open an issue for adding markup support for highlight and action items

**Review by fruch — APPROVED** [review]
LGTM

**Review by juliayakovlev — APPROVED** [review]

**Comment by soyacz** [comment]
@k0machi ping

**Comment by juliayakovlev** [comment]
Maybe it will be good emphasise which kind of error: bad performance or failure/error

**Comment by k0machi** [comment]
You can rebase to current master as I've merged #503 and then deploy it on staging.

File: argus_backend.py [hunk 1]
```diff
@@ -35,6 +36,7 @@ def start_server(config=None) -> Flask:
     app.register_blueprint(api.bp)
     app.register_blueprint(admin.bp)
     app.register_blueprint(cli_bp)
+    app.register_blueprint(highlights.bp)
```
[the remark this answers is not in the document]
Comment (soyacz) [reply argus_backend.py:39]: true, I added it under views blueprint (`/api/v1/views/widgets/highlights`)

**Review by k0machi — APPROVED** [review]

File: argus/backend/models/view_widgets.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    created_at = columns.DateTime(primary_key=True, clustering_order="DESC")
+    archived_at = columns.DateTime(default=datetime.fromtimestamp(0, tz=UTC))
+    creator_id = columns.UUID()
+    assignee_id = columns.UUID()
+    content = columns.Text()
+    completed = columns.Boolean(default=lambda: None)  # None means it's highlight, not an action item
+    comments_count = columns.TinyInt()
+
+class WidgetComment(Model):
+    view_id = columns.UUID(partition_key=True, required=True)
```
[the remark this answers is not in the document]
Comment (fruch) [reply argus/backend/models/view_widgets.py:19]: We should be able to put this widget on any view

We do want this collaboration for plans views

I did get the ephemeral part, on any update of plan, view is recreated ? so can it save the widgets changes ? If it doesn't, we should create a issue for it.

[the remark this answers is not in the document]
Comment (k0machi) [reply argus/backend/models/view_widgets.py:19]: It doesn't right now, the definitions for widgets are static. We would need to mark plan as edited and make plan updater to instead update existing view or re-use the original view id.

[the remark this answers is not in the document]
Comment (soyacz) [reply argus/backend/models/view_widgets.py:19]: > We should be able to put this widget on any view
> 

We can put it on any view, just upon plan change we replace whole view and thus this widget will not persist.

> We do want this collaboration for plans views
> 
> I did get the ephemeral part, on any update of plan, view is recreated ? so can it save the widgets changes ? If it doesn't, we should create a issue for it.

I propose to do it in followup where I'll discuss with @k0machi solutions. I think it won't require changes on model level, just providing widget settings should suffice.

**Pull request merged**